sdk/aqueduct/utils/naming.py

```diff
@@ -0,0 +1,101 @@
+from typing import List, Optional, Tuple, Union
+
+from aqueduct.models.dag import DAG
+from aqueduct.utils.dag_deltas import RemoveOperatorDelta, apply_deltas_to_dag
+
+
+def _artifact_name_from_op_name(op_name: str) -> str:
+    return op_name + " artifact"
+
+
+def _construct_default_output_artifact_names_from_op(op_name: str, num_outputs: int) -> List[str]:
+    """The default artifact naming policy is "<op_name> artifact (<optional counter>)".
+
+    In the multi-output case, we deduplicate each, starting the counter at 1.
+    """
+    artifact_names = [_artifact_name_from_op_name(op_name)]
+    if num_outputs == 1:
+        return artifact_names
+
+    artifact_names += [
+        _artifact_name_from_op_name(op_name) + " (%d)" % i for i in range(1, num_outputs)
+    ]
+    return artifact_names
+
+
+def resolve_op_and_artifact_names(
+    dag: DAG,
+    candidate_op_name: str,
+    overwrite_existing_op_name: bool,
+    candidate_artifact_names: Optional[Union[str, List[str]]] = None,
+    num_outputs: int = 1,
+    only_resolve_op_name: bool = False,
+) -> Tuple[str, List[str]]:
+    """Enforces our unique naming policy on operators and also their output artifacts.
+
+    Operator collisions are resolved. Artifact collisions are not tolerated.
+
+    Args:
+        dag:
+            The current dag.
+        candidate_op_name:
+            The operator name that we want to check and potentially deduplicate.
+        overwrite_existing_op_name:
+            If set, any operator collisions will be ignored. Otherwise, we will bump
```

Comment:
instead of "ignored", do you mean we overwrite the existing one?

Reply:
That's what I meant by ignored, but can change the wording around a bit.